src/App.js

```diff
@@ -4,9 +4,39 @@ import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export class App extends Component {
-  state = {};
+  state = {
+    movies: moviesFromServer,
+    query: '',
+  };
 
   render() {
+    let { movies } = this.state;
+    const { query } = this.state;
+
+    const appropTitles = movies.filter(film => (
+      film.title.toLowerCase().includes(query.toLowerCase())
+    ));
+
+    const appropDesc = movies.filter(film => (
+      film.description.toLowerCase().includes(query.toLowerCase())
+      && !film.title.toLowerCase().includes(query.toLowerCase())
+    ));
+
+    appropTitles.sort((film1, film2) => (
+      film1.description.toLowerCase().indexOf(query.toLowerCase())
+      - film2.description.toLowerCase().indexOf(query.toLowerCase())
+    ));
+
+    appropDesc.sort((film1, film2) => (
+      film1.description.toLowerCase().indexOf(query.toLowerCase())
```

Comment:
```suggestion
      film1.description.toLowerCase().indexOf(query.trim().toLowerCase())
```
убирай пробелы из query, они не должны влиять на результат